src/sql/workbench/services/tableDesigner/browser/tableDesignerComponentInput.ts

```diff
@@ -141,6 +141,7 @@ export class TableDesignerComponentInput implements DesignerComponentInput {
 	private doInitialization(designerInfo: azdata.designers.TableDesignerInfo): void {
 		this.updateState(true, false);
 		this._viewModel = designerInfo.viewModel;
+		this._originalViewModel = deepClone(this._viewModel);
```

Comment:
You should be updating this on save/apply as well, right?

Reply:
good catch